automation/routine_publish_signals.go

```diff
@@ -111,64 +49,70 @@ type PublishSignals struct {
 	Transforms []func(item *views.ItemSave)
 }
 
-// Do performs the automation against c with the passed in opts.
-func (p PublishSignals) Do(ctx context.Context, c *clarify.Client, opts PublishOptions) error {
-	var total int
+var _ Routine = PublishSignals{}
+
+func (p PublishSignals) Do(ctx context.Context, cfg *Config) error {
+	logger := cfg.Logger()
```

Comment:
we can have `.With("op", "PublishSignals.Do")` to know the source of logs

Reply:
Let's close and come back to this later. These logs are for the end-user, so including references to code may or may not be to helpful; especially not if custom routine functions must remember to do it themselves.